RestApi/Sepes.RestApi/Model/Study.cs

```diff
@@ -36,13 +39,52 @@ public class Study
             this(studyName, studyId, pods == null ? ImmutableHashSet<Pod>.Empty : pods, ImmutableHashSet<User>.Empty, 
                 ImmutableHashSet<User>.Empty, ImmutableHashSet<DataSet>.Empty, false, new int[]{}, new int[]{}) {}
 
-        public StudyInput ToStudyInput() // will be fully implemented later
+        public StudyInput ToStudyInput()
         {
+            var inputPods = new HashSet<PodInput>();
+            var inputSponsors = new HashSet<string>();
+            var inputSuppliers = new HashSet<string>();
+
+            foreach (Pod pod in pods) {
+                inputPods.Add(pod.ToPodInput());
+            }
```

Comment:
Tip: Can be simplified using Linq.  var inputPods = pods.Select(p => p.ToPodInput()).ToHashSet();
But may be a preference thing of loops vs iteration.